noxfile.py

```diff
@@ -19,23 +20,56 @@
 nox.options.sessions = ["lint", "docs"]
 
 
-# Files that need to be linted & formatted
-LINT_FILES = [
-    "ipynb-to-gallery.py",
-    "generate-gallery.py",
-    "noxfile.py",
-    "docs/src/conf.py",
-    "examples",
-    "developer",
-]
-
 EXAMPLES = get_examples()
 
 # ==================================================================================== #
 #                                 helper functions                                     #
 # ==================================================================================== #
 
 
+# Files that need to be linted & formatted
+def get_lint_files():
+    LINT_FILES = [
+        "ipynb-to-gallery.py",
+        "generate-gallery.py",
+        "noxfile.py",
+        "docs/src/conf.py",
+        "developer",
+    ]
+    return LINT_FILES
+
+
+def filter_files(tracked_files):
```

Comment:
I still don't understand why we have this function?

Reply:
there are files with are not ignored by git that I wanted to ignore by the linter. 
For ex the traj files or the yml files, also `git ls-files` contains also all the `.gitignore` files that i do not want to lint

Reply:
ah, because `git ls-files` gives you files which are not Python?

Reply:
Then I'm fine with this, maybe add some comment/docstring explaining the function?

Reply:
yes gives you everything is tracked by `git`
```
>>> git ls-files
.github/workflows/comment-docs-download.yml
.github/workflows/docs.yml
.github/workflows/lint.yml
.gitignore
CONDUCT.md
CONTRIBUTING.rst
LICENSE
README.rst
developer/get_examples.py
docs/requirements.txt
docs/src/.gitignore
docs/src/conf.py
docs/src/index.rst.in
examples/batch-cp2k/.gitignore
examples/batch-cp2k/README.rst
examples/batch-cp2k/environment.yml
examples/batch-cp2k/example.xyz
examples/batch-cp2k/reference-trajectory.py
examples/batch-cp2k/reftraj_template.cp2k
examples/batch-cp2k/run_calcs.sh
examples/dos-align/README.rst
examples/dos-align/dos-align.py
examples/dos-align/environment.yml
examples/gaas-map/.gitignore
examples/gaas-map/README.rst
examples/gaas-map/environment.yml
examples/gaas-map/gaas-map.py
examples/lode-linear/README.rst
examples/lode-linear/charge-charge.xyz
examples/lode-linear/environment.yml
examples/lode-linear/lode-linear.py
examples/lpr/.gitignore
examples/lpr/README.rst
examples/lpr/environment.yml
examples/lpr/lpr.py
examples/roy-gch/.gitignore
examples/roy-gch/README.rst
examples/roy-gch/environment.yml
examples/roy-gch/roy-gch.py
examples/sample-selection/.gitignore
examples/sample-selection/README.rst
examples/sample-selection/environment.yml
examples/sample-selection/input-fps.xyz
examples/sample-selection/sample-selection.py
generate-gallery.py
ipynb-to-gallery.py
noxfile.py
pyproject.toml
```